autorag/indexer/process/azure/table.py

```diff
@@ -0,0 +1,255 @@
+            # across pages.
+            if (
+                not has_headers
+                and last_table_headers
+                and col_count == last_table_col_count
+            ):
+                # Construct the table using last table headers and name
+                single_table = SingleTableProcessor(
+                    table_cells, last_table_name, last_table_headers
+                )
+                table_dataframes.append(single_table.table_dataframe)
+                table_pages.append(table_page)
+                continue
+
+            # Process normally if the table has headers
+            if has_headers:
+                if (
+                    not table_name
+                    and last_table_name
+                    and col_count == last_table_col_count
+                ):
+                    table_name = last_table_name
+
+                single_table = SingleTableProcessor(table_cells, table_name)
+                table_dataframes.append(single_table.table_dataframe)
+                table_pages.append(table_page)
+                # save the table headers, table name, and table column count
+                last_table_headers = single_table.headers
+                last_table_name = single_table.table_name
+                last_table_col_count = len(last_table_headers[0])
+
+        return table_dataframes, table_pages
+
+    def get_table_nodes(self) -> list[TextNode]:
+        """
+        Iterates through all tables stored in self.table_dataframes. For every
+        row in each table, a TextNode object is created. The text of the table
+        row is converted to a string and set as the text of the node.
+
+        :return: A list of TextNode objects each representing a row in the tables.
+        """
+        nodes = []
+        for idx in range(len(self.table_dataframes)):
+            table_df = self.table_dataframes[idx]
+            table_page = self.table_pages[idx]
+            for table_row_text in table_df:
+                # Create a new document for each page
+                new_node = TextNode(
+                    text=str(table_row_text),
+                    metadata={
+                        "page_number": table_page,
+                        "document_name": self.file_name,
+                        "document_type": DEFAULT_FILE_TYPE,
+                    },
+                )
+                nodes.append(new_node)
+        return nodes
+
+    def get_table_page_number(self, data: Dict[str, Any]) -> int:
+        """
+        Extracts the page number from table data.
+
+        :param data: Dictionary containing table and metadata.
+        :return: Page number of the table or 0 if not found.
+        """
+        bounding_regions = data.get("boundingRegions", [])
+        return bounding_regions[0].get("pageNumber", 0) if bounding_regions else 0
+
+
+class SingleTableProcessor:
+    """
+    Processes a single table data from a list of tables.
```

Comment:
can we handle complex tables where multiple cells are merged into a single cell?

Reply:
yes